### ISSUE TITLE
Work around for Firewall proxy while loading latest chrome driver

### ISSUE THREAD

**Issue by jawlitkp** [groupon/Selenium-Grid-Extras#179]
Hi ,

I am getting error while it tries to check latest version of chrome driver as it is talking to external URL and since i am in my company network ,it is preventing to get latest driver version,i telling script to not update my drivers but it is still checking for latest version,any work around ?

Thanks

**Comment by smccarthy** [comment]
I'll take a quick look at the code this weekend (after the holiday). If you have it set to not get the latest version, I don't think it should check for the latest version.

**Comment by arran-huxtable** [comment]
Yes it does, I get the same behaviour:

https://github.com/groupon/Selenium-Grid-Extras/blob/master/SeleniumGridExtras/src/main/java/com/groupon/seleniumgridextras/config/FirstTimeRunConfig.java#L233

As you can see in the code, regardless of whether I tell it to auto-update or not, it still goes off to the internet to find out the latest versions of stuff.

**Comment by jawlitkp** [comment]
Hello,

Any update on this ? 

Thanks

**Comment by cktech529** [comment]
I get the same behavior when I update to the latest builds of Grid Extras, it started when the URLs in the code changed from http to https (and I believe Java is throwing x509 certificate issues) A hack is to change the "https" back to "http" and I believe comment out the 1 test that fails on build... (So sync, modify and build your own copy perhaps?)

**Comment by jawlitkp** [comment]
Can you give fix for this issue please ,i can't even get started

**Comment by smccarthy** [comment]
@jawlitkp Is the FirstTimeRunConfig the problem for you? If so, are you looking for it to NOT download the chromedriver or IE driver or selenium-server-standalone.jar ?

FirstTimeRunConfig will currently always download those drivers, whether you want it to auto update the drivers or not. Maybe we should have an option to not download third party drivers. Is that what you are looking for?

Thank you!

Btw, someone mentioned a trick that uses an http and https proxy when running Selenium-Grid-Extras. Maybe that could help you?

"C:\Program Files\Java\jdk1.7.0_79\bin\java.exe" -Dhttp.proxyHost=yourproxy.com -Dhttp.proxyPort=8080 -Dhttps.proxyHost=yourproxy.com -Dhttps.proxyPort=8080 -cp "C:\selenium_ex/SeleniumGridExtras-jar-with-dependencies.jar;C:\selenium_ex/all-node-extensions-0.6.jar" -jar "C:\selenium_ex/SeleniumGridExtras-jar-with-dependencies.jar"

**Comment by mrdfuse** [comment]
@smccarthy yes I have the same problem. On a company network where we don't allow our build/test infrastructure to go reach out to the wild internet, and even if it could, it would encounter a mitm proxy/firewall with selfsigned certificates upon which all tools fail.

Problem occurs in FirstTimeRunConfig.setDriverAutoUpdater (rule 248). Even when answered 'no' on the question on automatic updating, it stills tries to download the latest version:

`14 okt 2016 14:32:23 [main] INFO  webdriverreleasemanager.WebDriverReleaseManager  - Checking the latest version of WebDriver, IEDriver, ChromeDriver and GeckoDriver from https://selenium-release.storage.googleapis.com/ and https://chromedriver.storage.googleapis.com/LATEST_RELEASE and https://api.github.com/repos/mozilla/geckodriver/releases
14 okt 2016 14:32:25 [main] ERROR config.RuntimeConfig  - Something went wrong loading webdriver versions
14 okt 2016 14:32:25 [main] ERROR config.RuntimeConfig  - org.dom4j.DocumentException: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target Nested exception: sun.security.validator.ValidatorException: PKIX path building failed: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target`

Here it is encountering our mitm proxy with selfsigned certificates (which I could solve by importing our rootCA in the JDK's truststore). Ideally there would be an option to just provide your own drivers without the jar having to go to the internet.

**Comment by theandrewlane** [comment]
@mrdfuse - Have you tried @smccarthy's workaround? That's what I've been using behind my corporate proxy.

**Comment by mrdfuse** [comment]
No because the machine I tried it on shouldn't be allowed to go to the internet, this was rectified meanwhile in our company.

**Comment by smccarthy** [comment]
@mrdfuse @theandrewlane I think we need a feature that asks the user if they want to download drivers or provide your own drivers. I don't think it would be too difficult to add. It would have to be added to the FirstTimeRunConfig.

**Comment by smccarthy** [comment]
I don't think it actually downloads the drivers if they already exist.

On OSX, chromedriver is : /tmp/webdriver/chromedriver/chromedriver_2.24_64bit
selenium-server-standalone is : /tmp/webdriver/3.0.0.jar (or 2.53.1.jar)
geckodriver is : /tmp/webdriver/geckodriver/geckodriver (I should really add the version to this)

**Comment by mrdfuse** [comment]
Hmm I'll need to debug for that to make sure. Will check later at work.

**Comment by glacier6327** [comment]
I am seeing this on Windows 10.... I tried install the cert wit keytool, but has no affect... What is the recommended process for moving forward with this.??

**Comment by mcafeerun** [comment]
Sorry for commenting on such a old issue.. I have similar situation where http calls to any external site is blocked from our VMs. So, while setting up the extras, i end up with failure.

Is there a way to supply the downloaded selenium server jar and drivers from my local directory to this configuration process?

Appreciate it!
